File: linear-search/linear-search.rb
def linear_search(arr, target)
  arr.each_with_index do |val, i|
    return i if val == target
  end
  -1
end

arr = [5, 2, 8, 1, 9, 3]
puts linear_search(arr, 8)
puts linear_search(arr, 7)

File: linear-search/linear-search.go
package main

import "fmt"

func linearSearch(arr []int, target int) int {
    for i, val := range arr {
        if val == target {
            return i
        }
    }
    return -1
}

func main() {
    arr := []int{5, 2, 8, 1, 9, 3}
    fmt.Println(linearSearch(arr, 8))
    fmt.Println(linearSearch(arr, 7))
}
